src/items/containers/mailbox/mailbox.cpp

```diff
@@ -72,49 +72,67 @@ void Mailbox::postRemoveNotification(const std::shared_ptr<Thing> &thing, const
 }
 
 bool Mailbox::sendItem(const std::shared_ptr<Item> &item) const {
+	if (!item) {
+		return false;
+	}
+
 	std::string receiver;
 	if (!getReceiver(item, receiver)) {
 		return false;
 	}
 
-	/**No need to continue if its still empty**/
+	/** No need to continue if it's still empty **/
 	if (receiver.empty()) {
 		return false;
 	}
 
+	Container* container = item->getContainer().get();
+	if (!container) {
+		return false;
+	}
+
 	if (item && item->getContainer() && item->getTile()) {
 		for (const auto &spectator : Spectators().find<Player>(item->getTile()->getPosition())) {
-			spectator->getPlayer()->autoCloseContainers(item->getContainer());
+			if (spectator && spectator->getPlayer()) {
+				spectator->getPlayer()->autoCloseContainers(item->getContainer());
```

Comment:
Here you can change to the "container". This will also be more performant, as it won't call "getContainer" every time the loop runs, just once.